### PR TITLE
Improved user locator

### PR DESCRIPTION
**Pull request by MarcoGuerrieriTAB**
Improved user locator use, removed the first location with coordinates 0,0 and implemented an authorization checker

### CODE REVIEW

File: TideApp/TideAppWidget/Domain/TidesWidgetDataProvider.swift [hunk 1]
```diff
@@ -53,17 +52,20 @@ class TidesWidgetDataProvider: NSObject, TidesWidgetDataProviderType, Observable
...
-      .sink(receiveCompletion: { result in
-        guard case .failure = result else { return }
-        completion(.failure(error: "Error on retrieving your current location"))
-      }, receiveValue: { location in
-        self.getDataFor(location: location, completion: completion)
+      .debounce(for: .milliseconds(500), scheduler: RunLoop.main)
+      .sink(receiveValue: { result in
+        switch result {
+        case .awaiting:
+          break // loading is authomated in widget
```
Comment (dsmith42) [inline TideApp/TideAppWidget/Domain/TidesWidgetDataProvider.swift:61]: Super minor just a typo in comment
```suggestion
          break // loading is automated in widget
```

Comment (MarcoGuerrieriTAB) [reply TideApp/TideAppWidget/Domain/TidesWidgetDataProvider.swift:61]: Thanks!!! 😄

**Review by dsmith42 — APPROVED** [review]
LGTM - Nice to add in the authorisation status check.

**Review by SophieClarkKC — APPROVED** [review]
Nice!

**Pull request merged**